sosw/components/dynamo_db.py

```diff
@@ -428,7 +428,8 @@ def get_by_query(self, keys: Dict, table_name: Optional[str] = None, index_name:
         result = []
 
         if return_count:
-            return sum([page['Count'] for page in response_iterator])
+            count = sum([page['Count'] for page in response_iterator])
```

Comment:
Don't you think that `return_count` and `max_items` are self-neglecting?
I would simply not allow both arguments simultaneously.